asserts/gpgkeypairmgr.go

```diff
@@ -183,3 +223,83 @@ func (gkm *GPGKeypairManager) sign(fingerprint string, content []byte) ([]byte,
 	}
 	return out, nil
 }
+
+type gpgKeypairInfo struct {
+	pubKey      PublicKey
+	fingerprint string
+}
+
+func (gkm *GPGKeypairManager) findByName(name string) (*gpgKeypairInfo, error) {
+	stop := errors.New("stop marker")
+	var hit gpgKeypairInfo
+	match := func(privk PrivateKey, fpr string, uid string) error {
+		if uid == name {
+			hit = gpgKeypairInfo{
+				pubKey:      privk.PublicKey(),
+				fingerprint: fpr,
+			}
+			return stop
+		}
+		return nil
+	}
+	err := gkm.Walk(match)
+	if err == stop {
+		return &hit, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return nil, fmt.Errorf("cannot find key named %q in GPG keyring", name)
+}
+
+var generateTemplate = `
+Key-Type: RSA
+Key-Length: 4096
+Subkey-Type: RSA
+Subkey-Length: 4096
+Name-Real: %s
+Preferences: SHA512
+`
+
+// Generate creates a new key with the given passphrase and name.
+func (gkm *GPGKeypairManager) Generate(passphrase string, name string) error {
+	_, err := gkm.findByName(name)
+	if err == nil {
+		return fmt.Errorf("key named %q already exists in GPG keyring", name)
+	}
+	generateParams := fmt.Sprintf(generateTemplate, name)
+	if passphrase != "" {
+		generateParams += "Passphrase: " + passphrase + "\n"
+	}
```

Comment:
I think with gpg2 we could ask it with a %command or it would default to ask for a passphrase on its own even with --batch